src/Config/Source/CloudSource.php

```diff
@@ -217,6 +218,22 @@ private function addPhp(
         ]);
 
         if ($extensions) {
+
+            /* Parse nested Blackfire configuration */
+            foreach ($extensions as $key => $phpExtension) {
+                if (is_array($phpExtension) && $phpExtension["name"] === ServiceInterface::SERVICE_BLACKFIRE) {
+                    /* Store extension data in repository */
+                    $repository->set([
+                        self::SERVICES_BLACKFIRE_ENABLED => true,
+                        self::SERVICES_BLACKFIRE_VERSION => $this->serviceFactory->getDefaultVersion(ServiceInterface::SERVICE_BLACKFIRE),
+                        self::SERVICES_BLACKFIRE_IMAGE => $this->serviceFactory->getDefaultImage(ServiceInterface::SERVICE_BLACKFIRE),
+                        self::SERVICES_BLACKFIRE_CONFIG => $phpExtension["configuration"]
+                    ]);
+                    /* Reset nested extension data with extension name */
+                    $extensions[$key] = ServiceInterface::SERVICE_BLACKFIRE;
+                }
+            }
+
             $repository[self::PHP_ENABLED_EXTENSIONS] = $extensions;
```

Comment:
Parse PHP extensions in `.magento.app.yaml` and if Blackfire is enabled, store Blackfire data and configuration in the repository. Finally, reset the nested data to a single `blackfire` string to make extension parsing later to be consistent and simple since we've already parsed and stored the Blackfire configuration.